src/main/java/cz/codecamp/logger/loggers/FileLogger.java

```diff
@@ -0,0 +1,78 @@
+package cz.codecamp.logger.loggers;
+
+import cz.codecamp.logger.*;
+import cz.codecamp.logger.fileHandlers.DayBasedFileHandler;
+import cz.codecamp.logger.fileHandlers.FileHandler;
+import cz.codecamp.logger.fileHandlers.IntervalBasedFileHandler;
+import cz.codecamp.logger.formatters.LogFormatter;
+
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.PrintStream;
+import java.time.temporal.ChronoUnit;
+
+public class FileLogger extends Logger implements LoggerInterface, PragmaticLoggerInterface {
+
+    private String fileName = "app.log";
+    private PrintStream fileStream;
+    private FileHandler fileHandler;
+    private FormatterInterface defaultFormatter;
+
+    public FileLogger() {
+
+        fileHandler = new IntervalBasedFileHandler(fileName, ChronoUnit.MINUTES, 1);
+        //fileHandler = new DayBasedFileHandler(fileName);
+        defaultFormatter = new LogFormatter();
+    }
+
+    public FileLogger(FileHandler fileHandler) {
+        this.fileHandler = fileHandler;
+        defaultFormatter = new LogFormatter();
+    }
+
+    private void adjustFileStream() {
+        if (fileHandler.hasLogFileLifeExpired()) {
+            if (fileStream!=null) {
+                fileStream.close();
```

Comment:
Pozor na multi-threading - pokud budu stejný logger volat z více threadů, tady mi to nejspíš failne, protože v jednom threadu zavřu stream, v druhém budu chtít něco zalogovat, ale nebudu mít ještě otevřen nový soubor. Chtělo by to použít nějaký synchronizační mechanismus.